### PR TITLE
add development ship boot option

### PR DESCRIPTION
**Pull request by h5gq3**

### CODE REVIEW

File: src/renderer/details/pages/DevshipDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        >
+            <h1 className="font-semibold text-base mb-4">Enter development ship details</h1>
+            <div>
+                <label htmlFor="name">Name <span className="text-gray-300 dark:text-gray-700">(local only)</span></label>
+                <NameField form={form} />
+            </div>
+            <div>
+                <label htmlFor="shipname">Shipname <span className="text-gray-300 dark:text-gray-700">(Urbit ID)</span></label>
+                <ShipNameField form={form} />
+            </div>
```
Comment (arthyn) [inline src/renderer/details/pages/DevshipDetails.tsx:31]: Can we add another field for extra command line args? I think it's pretty common to need those in dev especially for booting from a solid pill.

File: src/renderer/details/pages/ExistingShipDetails.tsx [hunk 2]
```diff
@@ -40,6 +40,7 @@ export const ExistingShipDetails = () => {
                     <option value="star">Star</option>
                     <option value="moon">Moon</option>
                     <option value="comet">Comet</option>
+                    <option value="devship">Development ship</option>
```
Comment (arthyn) [inline src/renderer/details/pages/ExistingShipDetails.tsx]:
```suggestion
                    <option value="devship">Development Ship</option>
```

File: src/renderer/routes.ts [hunk 3]
```diff
@@ -14,6 +14,14 @@ const comet: RouteInfo = {
     description: 'Generate a disposable identity and boot as a comet'
 }
 
+const devship: RouteInfo = {
+    key: 'devship',
+    path: '/boot/devship',
+    title: 'Development ship',
```
Comment (arthyn) [inline src/renderer/routes.ts]:
```suggestion
    title: 'Development Ship',
```

File: src/renderer/ship/components/BootOptions.tsx [hunk 4]
```diff
@@ -17,7 +17,7 @@ const BootLink = ({ route }: { route: RouteInfo }) => (
 export const BootOptions = ({ className = '' }: { className: string }) => {
     const [showMore, setShowMore] = useState(false);
     const mainMenu = [routeMap.planet, routeMap.comet];
-    const secondaryMenu = [routeMap.moon, routeMap.existing, routeMap.remote];
+    const secondaryMenu = [routeMap.devship, routeMap.moon, routeMap.existing, routeMap.remote];
```
Comment (arthyn) [inline src/renderer/ship/components/BootOptions.tsx]:
```suggestion
    const secondaryMenu = [routeMap.moon, routeMap.existing, routeMap.remote, routeMap.devship];
```

File: src/renderer/routes.ts [hunk 5]
```diff
@@ -69,6 +78,7 @@ export const routes: RouteInfo[] = [
     planet,
     moon,
     comet,
+    devship,
```
Comment (arthyn) [inline src/renderer/routes.ts:81]: I think dev ships should come last, it's less likely than the other options imo.

**Review by arthyn — CHANGES_REQUESTED** [review]
Mostly minor stuff, let me know if you need help adding the extra args

File: src/renderer/details/pages/DevshipDetails.tsx (hunk 1 above)
Comment (h5gq3) [reply src/renderer/details/pages/DevshipDetails.tsx:31]: added

File: src/background/services/pier-service.ts [hunk 6]
```diff
@@ -317,8 +317,9 @@ export class PierService {
         if (pier.status !== 'unbooted') {
             return;
         }
-        
-        const args = this.getSpawnArgs(pier);
+        let args = this.getSpawnArgs(pier)
+        args.unshift(pier.spawnArgs);
```
Comment (arthyn) [inline src/background/services/pier-service.ts:321]: We should probably only put these on the array if we actually have some

File: src/background/services/pier-service.ts [hunk 7]
```diff
@@ -317,8 +317,9 @@ export class PierService {
...
+        console.log(args)
```
Comment (arthyn) [inline src/background/services/pier-service.ts:322]: We spit these out later on so this should be removed

File: src/renderer/details/pages/DevshipDetails.tsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+            <div>
+                <label htmlFor="spawnArgs">Boot options <span className="text-gray-300 dark:text-gray-700">agruments</span></label>
```
Comment (arthyn) [inline src/renderer/details/pages/DevshipDetails.tsx:33]:
```suggestion
                <label htmlFor="spawnArgs">Boot Arguments</label>
```

File: src/renderer/details/pages/DevshipDetails.tsx [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+    const { isValid } = form.formState;
+
+    return (
+        <DetailsContainer
+            title="Development ship details"
+            buttonDisabled={!isValid}
+            onSubmit={form.handleSubmit(data => mutate(data))}
+        >
+            <h1 className="font-semibold text-base mb-4">Enter development ship details</h1>
```
Comment (arthyn) [inline src/renderer/details/pages/DevshipDetails.tsx:23]:
```suggestion
            <h1 className="font-semibold text-base mb-4">Enter Development Ship Details</h1>
```

File: src/renderer/details/pages/DevshipDetails.tsx [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    const {
+        form,
+        mutate
+    } = useAddPier(data => send('add-pier', { ...data, type: 'devship' }));
+
+    const { isValid } = form.formState;
+
+    return (
+        <DetailsContainer
+            title="Development ship details"
```
Comment (arthyn) [inline src/renderer/details/pages/DevshipDetails.tsx:19]:
```suggestion
            title="Boot a Development Ship"
```

File: src/renderer/routes.ts (hunk 5 above)
Comment (arthyn) [reply src/renderer/routes.ts:81]: Still need this to be last, after remote

**Review by arthyn — CHANGES_REQUESTED** [review]
Sorry this took so long to get back to. A few more small things, but I was able to test this so as soon as these are addressed we can merge it

File: src/renderer/details/pages/DevshipDetails.tsx [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+                <label htmlFor="name">Name <span className="text-gray-300 dark:text-gray-700">(local only)</span></label>
+                <NameField form={form} />
+            </div>
+            <div>
+                <label htmlFor="shipname">Shipname <span className="text-gray-300 dark:text-gray-700">(Urbit ID)</span></label>
+                <ShipNameField form={form} />
+            </div>
+            <div>
+                <label htmlFor="spawnArgs">Boot options <span className="text-gray-300 dark:text-gray-700">agruments</span></label>
+                <ArgumentsField form={form} />
```
Comment (arthyn) [inline src/renderer/details/pages/DevshipDetails.tsx:34]: You probably should provide a validator which will check to make sure they don't try to provide `-F` themselves. The error could be "Booting a development ship includes -F by default". The name field has an example of something like this.

**Pull request closed without merging**